src/plonk/prover.rs

```diff
@@ -27,23 +27,28 @@ impl<C: CurveAffine> Proof<C> {
     ) -> Result<Self, Error> {
         struct WitnessCollection<F: Field> {
             advice: Vec<Polynomial<F, LagrangeCoeff>>,
-            _marker: std::marker::PhantomData<F>,
+            aux: Vec<Polynomial<F, LagrangeCoeff>>,
         }
 
         impl<F: Field> Assignment<F> for WitnessCollection<F> {
             fn assign_advice(
                 &mut self,
-                wire: AdviceWire,
+                wire: Wire,
                 row: usize,
                 to: impl FnOnce() -> Result<F, Error>,
             ) -> Result<(), Error> {
-                *self
-                    .advice
-                    .get_mut(wire.0)
-                    .and_then(|v| v.get_mut(row))
-                    .ok_or(Error::BoundsFailure)? = to()?;
-
-                Ok(())
+                match wire {
+                    Wire::AdviceWire(idx) => {
+                        *self
+                            .advice
+                            .get_mut(idx)
+                            .and_then(|v| v.get_mut(row))
+                            .ok_or(Error::BoundsFailure)? = to()?;
+
+                        Ok(())
+                    }
+                    _ => return Err(Error::WireTypeError),
```

Comment:
Once you've reverted the unification of aux and advice wires, this error condition is no longer possible and so can be removed.